### ISSUE TITLE
Node isn't inclued in nested/combine modular pipeline

### ISSUE THREAD

**Issue by haijames** [kedro-org/kedro-viz#1134]
## Description
I created 2 modular pipelines (with namespace), there is a dataset that is an input of this modular and also is an output of the other one. Then I created a pipeline that combines both of them. When I visualize it, the node that represents the dataset above is not included in the *mother modular pipeline, it instead is placed alone outside and didn't connect to the modular

## Context
I'm trying to group the modular pipelines so that the visualization look more neatly since our ETL process not just has a large number of steps but also has so many datasets that go through different process. I'm still new with kedro so maybe my code is not correct or maybe there is a solution somewhere on the Internet. If so, I'm sorry to trouble you guys and really appreciate if you guys can share it with me

## Steps to Reproduce

Here is a simple example of a pipelines that I'm trying to make

```
def create_pipeline(**kwargs) -> Pipeline:

    __root__ = pipeline(
        [
            node(
                func=transform_table_E,
                inputs=['Start'],
                outputs=['E'],
                name="transform_table_E",
            ),

        ],
    )
    # return __root__
    transform = pipeline(
        [
            node(
                func=filter,
                inputs=['A'],
                outputs=["A temp"],
                name="filter",
            ),
            node(
                func=join_A_B,
                inputs=['B', "A temp"],
                outputs=['C'],
                name="join_A_B",
            ),
        ],
        namespace="transform",
        inputs={'B', 'A' },
        outputs={'C'}
    )

    final = pipeline(
        [
            node(
                func=join_E_F,
                inputs=['E', 'F'],
                outputs=['D Temp'],
                name="join_E_F",
            ),
            node(
                func=join_C_D,
                inputs=['C', 'D Temp'],
                outputs=['D'],
                name="join_C_D",
            ),
        ],
        namespace="finally",
        inputs={'C', 'F', 'E'},
        outputs={'D'}
    )

    combine = pipeline(
        pipe= transform + final,
        namespace="train_evaluation",
        inputs={'E','A','B','F'},
    )

    return combine + __root__
```

To make it simpler, the content of the function is only  return True

```
def filter(A0,):
    return True

def transform_table_E(A0,):
    return True

def join_E_F(A0,A1,):
    return True

def join_A_B(A0,A1,):
    return True

def join_C_D(A0,A1,):
    return True 
```

and these are the simple catalog

```
Start:
  type: pandas.ExcelDataSet
  filepath: ''

A:
  type: pandas.ExcelDataSet
  filepath: ''

A temp:
  type: pandas.ExcelDataSet
  filepath: ''

B:
  type: pandas.ExcelDataSet
  filepath: ''

C:
  type: pandas.ExcelDataSet
  filepath: ''

D:
  type: pandas.ExcelDataSet
  filepath: ''

E:
  type: pandas.ExcelDataSet
  filepath: ''

F:
  type: pandas.ExcelDataSet
  filepath: '' 
```

## Expected Result
Still, this is not the true result that I expected, but it's correct for the picture, I got this result when I remove the namespace of the transform pipeline in an attempt to find the answer

![image](https://user-images.githubusercontent.com/45039771/195424510-1f999a23-adab-435f-9d06-b0a6a678d703.png)

You can see the dataset C is still not included in `Train Evaluation`, but somehow the visualization is what I expected, when I expanded it, the dataset C is somehow included in the `Train Evaluation`, you can see there are 2 dataset C in the pic below

![image](https://user-images.githubusercontent.com/45039771/195424392-60aa3641-0708-4feb-b24a-826ad988007a.png)

## Actual Result

This is what kedro viz shows when you applied the code I give above, the dataset C is alone and cries when he is not even connected to his brothers, but in the paper, C still belongs to the house `Train Evaluation`

![image](https://user-images.githubusercontent.com/45039771/195421562-e789d93b-8555-429d-a5ae-b816cba43b28.png)

But when you expand the `Train Evaluation` , the dataset C is connected

![image](https://user-images.githubusercontent.com/45039771/195421881-f948aa41-85ba-4f36-856c-6dc8d904e949.png)

![image](https://user-images.githubusercontent.com/45039771/195421849-12c74681-defa-44ba-ae83-5f58a0e617ff.png)

I think the main reason is that dataset C belongs to 2 different modular pipelines, so `kedro viz` is confused and doesn't know where to put it in the mother modular, so maybe it is not a bug report but a feature request 

For the record, I have tried many variations of either explicitly specifying inputs/outputs in the pipeline or implicitly relying on the automatic renaming of nodes, I also somehow removed the namespace in each modular in hope that can help. I always end up with the above visualization or worse. Also, if I instead remove the namespace of the `final` pipeline, the graph is missing and disconnected, not like when I remove the namespace of the `transform` pipeline

![image](https://user-images.githubusercontent.com/45039771/195423090-81730121-5e3e-4822-b8e4-2b348de12371.png)

## Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:

* Web browser system and version: Chrome
* Operating system and version: Windows 10 pro 21H2
* kedro 0.18.3
* kedro-viz 5.1.1
* Python version used: 3.9

## Checklist
- [x] Include labels so that we can categorise your issue

**Comment by tynandebold** [comment]
Thanks for this @haijames! This related to this issue that we're tracking here #1105. We're talking about how to solve this right now, so I'll come back with an update when I have one.

**Comment by tynandebold** [comment]
Hi @haijames, sorry for the long delay on this one.

I believe this issue was just fixed with the merge of #1439.

Here's the original problem:

![old](https://github.com/kedro-org/kedro-viz/assets/16869061/a2d681cb-5aff-4dad-92fa-1c04eeddd0a8)


And here it is with after the merge: 

![image](https://github.com/kedro-org/kedro-viz/assets/16869061/0fdf3d8e-49e7-4400-8b6e-2a8cbd12919f)

How does that look to you?

cc @rashidakanchwala

**Comment by tynandebold** [comment]
Closed via https://github.com/kedro-org/kedro-viz/pull/1439.
